### PR TITLE
Task3

### PR DESCRIPTION
**Pull request by paulius27**
Creates application that displays the weather of a user selected city.

### CODE REVIEW

File: WeatherApp/DAL/Models/OpenWeather/Main.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using System.Text.Json.Serialization;
+
+namespace DAL.Models.OpenWeather
+{
+    public class Main
```
Comment (MaximGomon) [inline WeatherApp/DAL/Models/OpenWeather/Main.cs]: Not a good idea to use Main as model name

File: WeatherApp/DAL/Models/OpenWeather/Wind.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Text.Json.Serialization;
+
+namespace DAL.Models.OpenWeather
+{
+    public class Wind
+    {
+        [JsonPropertyName("speed")]
```
Comment (MaximGomon) [inline WeatherApp/DAL/Models/OpenWeather/Wind.cs]: Json serialization\deserialization is case insensitive. I may skip attribute if difference is just a case

File: WeatherApp/DAL/WeatherRepository.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            _apiKey = apiKey;
+        }
+
+        public async Task<double> GetTemperatureByCityNameAsync(string cityName)
+        {
+            using HttpResponseMessage response = await _httpClient.GetAsync($"https://api.openweathermap.org/data/2.5/weather?q={cityName}&units=metric&APPID={_apiKey}");
+            response.EnsureSuccessStatusCode();
+
+            string responseBody = await response.Content.ReadAsStringAsync();
+            Models.OpenWeather.WeatherResponse weather = JsonSerializer.Deserialize<Models.OpenWeather.WeatherResponse>(responseBody);
```
Comment (MaximGomon) [inline WeatherApp/DAL/WeatherRepository.cs]: Please add using Models.OpenWeather; to prevent long type names

**Review by MaximGomon — CHANGES_REQUESTED** [review]

File: WeatherApp/DAL/Models/OpenWeather/Main.cs (hunk 1 above)
Comment (paulius27) [reply WeatherApp/DAL/Models/OpenWeather/Main.cs]: I renamed model to "MainWeatherData".

File: WeatherApp/DAL/Models/OpenWeather/Wind.cs (hunk 2 above)
Comment (paulius27) [reply WeatherApp/DAL/Models/OpenWeather/Wind.cs]: JsonPropertyName attributes have been removed.

File: WeatherApp/DAL/WeatherRepository.cs (hunk 3 above)
Comment (paulius27) [reply WeatherApp/DAL/WeatherRepository.cs]: I added "using" to prevent long type names.

**Review by MaximGomon — APPROVED** [review]

**Review by MaximGomon — APPROVED** [review]

**Review by MaximGomon — APPROVED** [review]

**Pull request merged**